### PR TITLE
change CI cluster trigger to use dbt debug

### PR DESCRIPTION
**Pull request by 0xRobin**
~~change CI steps order.
Compile needs to happen before the cluster trigger step, otherwise the cluster trigger step fails on compile errors and repeats for an hour.~~

Change the CI cluster trigger step to use `dbt debug`.
This saves us the additional compile time and also resolves any stuck-in-loop scenario's when having compile errors.

### CODE REVIEW

**Comment by aalan3** [comment]
doesn't `dbt run` also compile? not sure I understand why this is failing

**Comment by 0xRobin** [comment]
> EDIT: Ahh I think I get it, if there is a compile error you'd be stuck on the loop.

indeed. The compile error on the trigger run gets treated the same as 'cluster is not yet available, let's try again'

**Comment by 0xRobin** [comment]
@jeff-dude to validate this we should run the CI on this PR when the cluster is down. to make sure the compile step doesn't fail when it can't reach the cluster. I'm not sure anymore if the CI compile is dependent on the connection (as I can compile locally without one)

**Comment by 0xRobin** [comment]
Ok after slack convo, I'll adapt the step to use `dbt debug` instead of running the trigger model. 
dbt debug does a connection test which runs `select 1 as id` under the hood.
This will save us a compile (and thus ~2min) on every CI run.

File: .github/workflows/dbt_slim_ci.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
 
       - name: dbt dependencies
         run: "dbt deps"
-      
+
       - name: Run dbt model on Trino if dunesql engine is used, and retry until cluster is available
```
Comment (belen-pruvost) [inline .github/workflows/dbt_slim_ci.yml:60]:
```suggestion
      - name: Run dbt debug on Trino if dunesql engine is used, and retry until cluster is available
```

**Review by belen-pruvost — APPROVED** [review]
I've re run the CI checks of this PR and I can see that the `select 1 as id` query is being correctly routed on the spellbook-ci UI. So this works 🎉

**Comment by jeff-dude** [comment]
thanks @0xRobin for simplifying!

i will close this, as #4574 does this change and also adds to other workflows in the repo.

**Pull request closed without merging**